### PR TITLE
doc #20466 Java range-directives examples added

### PR DESCRIPTION
**Pull request by gosubpl**
- added Java helper for unmarshalling into Multipart.ByteRanges
- added Java example for range-directives

### CODE REVIEW

**Comment by gosubpl** [comment]
Refs #20466

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
ADD TO WHITELIST

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/RangeDirectivesExamplesTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        response.assertHeaderKindNotExists("Content-Range");
+
+        final CompletionStage<ArrayList<Multipart.ByteRanges.BodyPart>> completionStage =
+                response.entity(Unmarshaller.entityToMultipartByteRanges()).getParts()
+                        .runFold(new ArrayList<>(), (acc, n) -> {
+                            acc.add(n);
+                            return acc;
+                        }, materializer);
+        try {
+            final ArrayList<Multipart.ByteRanges.BodyPart> bodyParts =
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/RangeDirectivesExamplesTest.java]: just `List`?

**Comment by ktoso** [comment]
LGTM, thanks!

**Comment by gosubpl** [comment]
yes! just `List` apart from one place when implementation not interface is needed

**Comment by ktoso** [comment]
Thanks :) Will merge after validation

**Comment by ktoso** [comment]
Merged as promised, thanks a lot!

**Comment by gosubpl** [comment]
Thanks Konrad! And am very happy to to have this in time for 2.4.8 :)

**Comment by ktoso** [comment]
Yeah, me as well. Thanks a lot for being awesome about those PRs :)

**Pull request merged**